portality/models/journal.py

```diff
@@ -838,6 +838,20 @@ def language_name(self):
         if self.language is not None:
             langs = self.language
         langs = [datasets.name_for_lang(l) for l in langs]
+
+        # now we need to ensure that these are all correctly unicoded
+        def to_utf8_unicode(val):
+            if isinstance(val, unicode):
+                return val
+            elif isinstance(val, basestring):
+                try:
+                    return val.decode("utf8", "strict")
+                except UnicodeDecodeError:
+                    raise ValueError(u"Could not decode string")
```

Comment:
This could error out quite easily, but we'll catch it when it gets reported I guess.